### ISSUE TITLE
using naive bayes on titanic dataset

### ISSUE THREAD

**Issue by parsaanooshfar-lab** [parsaanooshfar-lab/homework-machine-learning#17]
می توانیم یک مثال با **Naive Bayes** روی دیتاست دیگری انجام دهیم. برای مثال، دیتاست **Titanic** که هدفش پیش‌بینی **نجات‌یافتن یا جان باختن مسافر** است.

در ادامه کد کامل همراه با محاسبه شاخص‌های ارزیابی و نمودارها ارائه می‌کنم:

```python
# کتابخانه‌ها
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

# بارگذاری دیتاست Titanic
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
data = pd.read_csv(url)

# بررسی داده‌ها
print(data.head())

# انتخاب ویژگی‌ها و برچسب
# برای سادگی فقط برخی ویژگی‌ها استفاده می‌شوند
X = data[["Pclass", "Sex", "Age", "SibSp", "Parch", "Fare"]]
y = data["Survived"]

# تبدیل داده‌های متنی به عددی
X = pd.get_dummies(X, columns=["Sex"], drop_first=True)

# جایگزینی مقادیر گمشده (NaN) در Age با میانگین
X["Age"].fillna(X["Age"].mean(), inplace=True)

# تقسیم داده به train و test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# ساخت مدل Naive Bayes
model = GaussianNB()
model.fit(X_train, y_train)

# پیش‌بینی
y_pred = model.predict(X_test)

# محاسبه شاخص‌های ارزیابی
accuracy = accuracy_score(y_test, y_pred)
conf_matrix = confusion_matrix(y_test, y_pred)
class_report = classification_report(y_test, y_pred)

print(f"Accuracy: {accuracy:.4f}")
print("\nConfusion Matrix:")
print(conf_matrix)
print("\nClassification Report:")
print(class_report)

# نمایش Confusion Matrix بصورت نموداری
plt.figure(figsize=(6,4))
sns.heatmap(conf_matrix, annot=True, fmt="d", cmap="Blues", xticklabels=[0,1], yticklabels=[0,1])
plt.xlabel("Predicted")
plt.ylabel("Actual")
plt.title("Confusion Matrix - Titanic Dataset")
plt.show()
```

### توضیح کد:

1. دیتاست Titanic برای پیش‌بینی **نجات یا عدم نجات مسافر** استفاده شده است.
2. برخی ویژگی‌ها مثل سن، کلاس سفر، تعداد همراهان، جنسیت و کرایه انتخاب شده‌اند.
3. ویژگی متنی `Sex` به عددی تبدیل شده است.
4. **Naive Bayes (GaussianNB)** برای مدل‌سازی استفاده شد.
5. شاخص‌های ارزیابی شامل **Accuracy، Confusion Matrix، Precision، Recall و F1-Score** هستند و Confusion Matrix هم به صورت نمودار نمایش داده شد.

---
